### ISSUE TITLE
Access denied in center of interest thumbnail

### ISSUE THREAD

**Issue by joshuaavalon** [yonjah/piwigo-privacy#5]
As title. After activating the plugin, `Photo sizes with crop` in Center of interest throw 403 error.

**Comment by joshuaavalon** [comment]
After turning on debug, the exception is 
```
Fatal error: Uncaught Exception: Could not validate path (./i.php?/upload/2017/04/25/20170425004429-8c72dc1b-sq.png) actually belong to image (7) in /media/web/piwigo/plugins/piwigo_privacy/helper_funcs.inc.php:25 Stack trace: #0 /media/web/piwigo/plugins/piwigo_privacy/helper_funcs.inc.php(244): pwg_privacy_error('Could not valid...') #1 /media/web/piwigo/plugins/piwigo_privacy/get.php(58): pwg_privacy_verify_access(7, './i.php?/upload...') #2 {main} thrown in /media/web/piwigo/plugins/piwigo_privacy/helper_funcs.inc.php on line 25
```
After check the url, the crop image in the site is 
```
file=_data/i/upload/2017/04/25/20170425004429-8c72dc1b-cu_e120.png
```
but the center of interest thumbnail url is 
```
file=i.php?/upload/2017/04/25/20170425004429-8c72dc1b-sq.png&b=1493129013
```

**Comment by joshuaavalon** [comment]
It seems that it is a bug in `picture_coi.php`, row 70
```
  if ($conf['derivative_url_style']==1)
  {
    $conf['derivative_url_style']=0; //auto
  }
```

**Comment by joshuaavalon** [comment]
Open a issue there, leave a link for reference https://github.com/Piwigo/Piwigo/issues/659

**Comment by yonjah** [comment]
Thanks I'll follow the discussion there

**Comment by yonjah** [comment]
Is this still an issue for you ?
I'm not sure how to get the 'suqare' size to work. it seems like it's enabled by default but I don't have the size option on the image view, so I'm not sure how to reproduce this

**Comment by joshuaavalon** [comment]
It is an issue because I cannot see the thumbnail after changing center of interest. Currently I comment out the the script and it works fine.

I think  you can try to parse the url with `i.php?`

**Comment by yonjah** [comment]
Maybe I can find a way to bypas that.
But I really need to reproduce the issue   first. I tried changing the coi
but I couldn't reproduce this issue

**Comment by joshuaavalon** [comment]
What is the thumbnail url in your coi?

**Comment by joshuaavalon** [comment]
For reference, I use the latest Piwigo,  with [Darkroom](https://github.com/tkuther/piwigo-bootstrap-darkroom) theme  and your plugin only.

**Comment by yonjah** [comment]
I added a small fix in `0.1.1` to rewrite i.php if it exist.
It should fix this issue

**Comment by joshuaavalon** [comment]
Just tested it. It works perfectly now 😘
